dolphin/task_manager/manager.py

```diff
@@ -67,19 +70,38 @@ def say_hello(self, context, request_spec=None,
             pass
 
     def sync_storage_resource(self, context, storage_id, resource_task):
-        LOG.debug("Received the sync_storage task: {0} request for storage"
-                  " id:{1}".format(resource_task, storage_id))
-        cls = importutils.import_class(resource_task)
-        device_obj = cls(context, storage_id)
-        device_obj.sync()
+        lock = coordination.Lock(storage_id + resource_task)
+        if lock.acquire(False):
+            LOG.debug("Received the sync_storage task: {0} request for storage"
+                      " id:{1}".format(resource_task, storage_id))
+            cls = importutils.import_class(resource_task)
+            device_obj = cls(context, storage_id)
+            device_obj.sync()
+            lock.release()
+        else:
+            LOG.info("%s is rejected for %s because "
+                     "task is already running" % (resource_task, storage_id))
 
     def remove_storage_resource(self, context, storage_id, resource_task):
-        cls = importutils.import_class(resource_task)
-        device_obj = cls(context, storage_id)
-        device_obj.remove()
+        lock = coordination.Lock(storage_id + resource_task)
+        if lock.acquire(False):
+            cls = importutils.import_class(resource_task)
```

Comment:
Suggest to add a log when deleting resource.

Reply:
Done